src/advenjure/items.cljc

```diff
@@ -29,16 +29,25 @@
 
 (defn talk-defaults [item] (if (:dialog item) {:talk true}))
 
+(defn apply-synonyms
+  [item]
+  (let [keys (keys item)]
+    (map (fn [field]
```

Comment:
look into the `reduce` function, that's what you want to use here instead of `map`. You walk through each key building a new map: if the value is a key in the old map, use the value it points to, otherwise keep the current one. Something in the lines of `(assoc new-map key (get old-map value value))`.

BTW you don't want to be using stuff like `.indexOf`, clojure has great manipulation functions that you can take advantage on. [This might help](https://clojure.org/api/cheatsheet).